tmk_core/common/keyboard.c

```diff
@@ -336,32 +341,37 @@ void keyboard_task(void) {
     matrix_scan();
 #endif
 
-    if (should_process_keypress()) {
-        for (uint8_t r = 0; r < MATRIX_ROWS; r++) {
-            matrix_row    = matrix_get_row(r);
-            matrix_change = matrix_row ^ matrix_prev[r];
-            if (matrix_change) {
+    for (uint8_t r = 0; r < MATRIX_ROWS; r++) {
+        matrix_row    = matrix_get_row(r);
+        matrix_change = matrix_row ^ matrix_prev[r];
+        if (matrix_change) {
 #ifdef MATRIX_HAS_GHOST
-                if (has_ghost_in_row(r, matrix_row)) {
-                    continue;
-                }
+            if (has_ghost_in_row(r, matrix_row)) {
+                continue;
+            }
 #endif
-                if (debug_matrix) matrix_print();
-                matrix_row_t col_mask = 1;
-                for (uint8_t c = 0; c < MATRIX_COLS; c++, col_mask <<= 1) {
-                    if (matrix_change & col_mask) {
+            if (debug_matrix) matrix_print();
+            matrix_row_t col_mask = 1;
+            for (uint8_t c = 0; c < MATRIX_COLS; c++, col_mask <<= 1) {
+                if (matrix_change & col_mask) {
+                    if (should_process_keypress()) {
                         action_exec((keyevent_t){
                             .key = (keypos_t){.row = r, .col = c}, .pressed = (matrix_row & col_mask), .time = (timer_read() | 1) /* time should not be 0 */
                         });
-                        // record a processed key
-                        matrix_prev[r] ^= col_mask;
+                    }
+                    // record a processed key
+                    matrix_prev[r] ^= col_mask;
+
+#if defined(RGB_MATRIX_ENABLE)
+                    process_rgb_matrix(r, c, (matrix_row & col_mask));
```

Comment:
I feel the naming for this function should be changed, since it's not longer part of the `process_*` chain. 

But, `rgb_matrix_matrix_handler` sounds like a horrendous name, and I can't think of a better one.

Reply:
rgb_matrix_handle_matrix?

Reply:
Sounds just as horrible as my suggestion.  somebody may have a better name, but... for now, I think it's okay.